packages/apputils/src/clientsession.ts

```diff
@@ -596,16 +596,20 @@ class ClientSession implements IClientSession {
    */
   private _startSession(model: Kernel.IModel): Promise<Kernel.IKernelConnection> {
     if (this.isDisposed) {
-      return Promise.resolve(void 0);
+      return Promise.reject('Session is disposed.');
     }
     return this.manager.startNew({
       path: this._path,
       type: this._type,
       name: this._name,
       kernelName: model ? model.name : null,
       kernelId: model ? model.id : null
-    }).then(session => this._handleNewSession(session))
-    .catch(err => this._handleSessionError(err));
+    }).then(session => {
+      return this._handleNewSession(session);
+    }).catch(err => {
+      this._handleSessionError(err);
+      return this._session.kernel;
```

Comment:
I'm assuming we should return the original kernel here, rather than returning `undefined` which would cause the type of the return promise to be `Promise<IKernelConnection | undefined>` and force all downstream consumers to handle the `undefined` case.